spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2220,6 +2220,16 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
               expr.children(2))
             None
           }
+        case Size(child, _) if child.dataType.isInstanceOf[ArrayType] =>
```

Comment:
This either needs to handle when `legacySizeOfNull` is true or pattern match to only false